File: JS/QuemPagaraConta.test.js
/**
 * @version JAVASCRIPT
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./QuemPagaraConta.js')

TEST('Starting QuemPagaraConta test...', (t) => {
    //Test 1
    let want = [1, 1, 1, 0]
    let got = OBJ.resultOfSum( [3, 11, 7, 18] )
    t.assert(want.toString === got.toString, "Expect: 1, 1, 1, 0")
    //Test 2
    want = [1, 1, 1, 1, 1, 1, 1, 1]
    got = OBJ.resultOfSum( [7, 1, 1, 3, 3, 1, 7, 99] )
    t.assert(want.toString === got.toString, "Expect: 1, 1, 1, 1, 1, 1, 1, 1")
    //Test 3
    want = [0, 0, 0, 0, 0, 0, 0]
    got = OBJ.resultOfSum( [6, 2, 2, 4, 8, 10, 128] )
    t.assert(want.toString === got.toString, "Expect: 0, 0, 0, 0, 0, 0, 0")

    t.end()
})

File: JAVA/QuemPagaraContaTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertArrayEquals;
import org.junit.Test;

public class QuemPagaraContaTest {

	@Test
	public void testQuemPagaraConta() throws Exception {
		//Test 1
		int[] want = {1, 1, 1, 0};
		int[] got = QuemPagaraConta.resultOfSum( new int[] {3, 11, 7, 18} );
        assertArrayEquals(want, got);
		//Test 2
		want = new int[] {1, 1, 1, 1, 1, 1, 1, 1};
		got = QuemPagaraConta.resultOfSum( new int[] {7, 1, 1, 3, 3, 1, 7, 99} );
        assertArrayEquals(want, got);
		//Test 3
		want = new int[] {0, 0, 0, 0, 0, 0, 0};
		got = QuemPagaraConta.resultOfSum( new int[] {6, 2, 2, 4, 8, 10, 128} );
        assertArrayEquals(want, got);
	}

}
